spring-integration-redis/src/test/java/org/springframework/integration/redis/rules/RedisAvailableTests.java

```diff
@@ -85,6 +92,26 @@ protected void awaitContainerSubscribedWithPatterns(RedisMessageListenerContaine
 		Thread.sleep(1000);
 	}
 
+	protected void awaitFullySubscribed(RedisMessageListenerContainer container, RedisTemplate<?, ?> redisTemplate,
+			String redisChannelName, QueueChannel channel, Object message) throws Exception {
+		awaitContainerSubscribedNoWait(container);
+		drain(channel);
+		long now = System.currentTimeMillis();
+		Message<?> received = null;
+		while (received == null && System.currentTimeMillis() - now < 10000) {
+			redisTemplate.convertAndSend(redisChannelName, message);
+			received = channel.receive(1000);
+		}
+		drain(channel);
+		assertNotNull("Container failed to fully start", received);
```

Comment:
This solution smells a bit, because It is intended the same goal as original test-case. And what it is bad it doesn't help for other "race-conditional" test. So, it looks better to more the solution to the target. But from other side Redis "bad" stuff so often waste our time that I decided to merge you change as is.